File: app/src/main/java/com/tony/builder/game2048/util/DumpUtils.java
package com.tony.builder.game2048.util;

public class DumpUtils {
    int[][] currentMap = {
            {0,0,0,0},
            {2,2,0,0},
            {2,0,2,0},
            {0,2,2,0}
    };
    public static String dump(int[][] array) {
        StringBuilder sb = new StringBuilder();
        sb.append("{\n");
        for(int i=0; i<array.length; i++) {
            sb.append("\t");
            sb.append("{");
            for(int j=0; j<array[i].length; j++) {
                sb.append(array[i][j]);
                if ((j+1) != array[i].length) {
                    sb.append(",");
                }
            }
            sb.append("}\n");
        }
        sb.append("}\n");
        return sb.toString();
    }
}

File: app/src/main/java/com/tony/builder/game2048/util/DumpUtils.kt
package com.tony.builder.game2048.util

object DumpUtils {
    fun isEqual(expect: Array<IntArray>, value: Array<IntArray>): Boolean {
        if (expect.size != value.size) {
            return false
        }
        for (i in expect.indices) {
            if (value[i].size != expect[i].size) {
                return false
            }
            for (j in expect[i].indices) {
                if (value[i][j] != expect[i][j]) {
                    return false
                }
            }
        }
        return true
    }

    fun dump(array: Array<IntArray>): String {
        val sb = StringBuilder()
        sb.append("{\n")
        for (i in array.indices) {
            sb.append("\t")
            sb.append("{")
            for (j in array[i].indices) {
                sb.append(array[i][j])
                if (j + 1 !in array[i].indices) {
                    sb.append(",")
                }
            }
            sb.append("}\n")
        }
        sb.append("}\n")
        return sb.toString()
    }
}